metric_learn/mlkr.py

```diff
@@ -98,14 +98,14 @@ def _loss(flatA, X, y, dX):
   A = flatA.reshape((-1, X.shape[1]))
   dist = pdist(X, metric='mahalanobis', VI=A.T.dot(A))
   K = squareform(np.exp(-dist**2))
+  np.fill_diagonal(K, 0)
```

Comment:
I believe we need to set `K`'s diagonal to 0 here to prevent taking into account `Kii` in the sum called `denom`, and in the dot product with `y` (for the cost function) (and though it is involved in the gradient, it should still be correct since whatever the diagonal of `K` is in the gradient, it will be canceled out by zeros in `dX`)